lib/rules/no-smart-quotes.js

```diff
@@ -41,29 +49,47 @@ module.exports = {
                     return;
                 }
 
-                const match = node.raw.match(/^(.*(“|”|‘|’).*)$/);
-
-                if (match) {
-                    const offendingPunctuation = match[2];
-                    const matchedString = match[1];
-                    const alternate = {
-                        '“': '"',
-                        '”': '"',
-                        '‘': "'",
-                        '’': "'",
-                    } [offendingPunctuation];
-
-                    if (typeof matchedString !== 'undefined' && typeof offendingPunctuation !== 'undefined') {
-                        const message = `Strings must use straight quotes in: ${ matchedString}. Use ${alternate} instead.`;
-                        context.report({
-                            node,
-                            message: message,
-                            data: {
-                                offendingPunctuation
-                            }
-                        });
-                    }
+                const matches = Array.from(matchAll(node.raw, SMART_QUOTES_REGEX));
+
+                if (matches.length === 0) {
+                    return;
                 }
+
+                context.report({
+                    node,
+                    message: "Strings must use straight quotes.",
+                    fix(fixer) {
+                        const stringQuoteType = node.raw.charAt(0);
+
+                        let fixed = ""
+                        let matchIndex = 0;
+
+                        for (let index = 0; index < node.raw.length; index++) {
+                            let char = node.raw.charAt(index);
+                            const match = matches[matchIndex];
+
+                            if (match && index === match.index) {
+                                matchIndex++;
+
+                                let replacement = REPLACEMENTS[match[0]];
+
+                                if (replacement) {
+                                    if (replacement === stringQuoteType) {
+                                        replacement = '\\' + replacement;
```

Comment:
Need to escape here so that it doesn't replace `"“"` with `"""` (invalid syntax)